### PR TITLE
252 add frontend support for updating events

### PR DESCRIPTION
**Pull request by Frank-D-stein**
# Description
Added frontend support for updating events.

## Relevant issue(s)
- #252 

## Questions
no questions

## Type of change
- [x] New feature (non-breaking change which adds functionality)

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have assigned reviewers to this PR

### CODE REVIEW

File: src/app/events/[eventId]/edit/page.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import { EventResponse } from '@/types/dataModel/event';
+import EditEventView from '@/views/EditEventView';
+
+export default async function EventPage({
+  params,
+}: {
+  params: { eventId: string };
+}) {
+  const event: EventResponse = JSON.parse(
+    JSON.stringify(await getEvent(params.eventId))
```
Comment (zaviermiller) [inline src/app/events/[eventId]/edit/page.tsx:11]: dont need to do the `JSON.parse(JSON.stringify(...))`

File: src/types/forms/events.ts [hunk 2]
```diff
@@ -9,6 +9,18 @@ export const zTime = z.string().regex(/^([01]?[0-9]|2[0-3]):[0-5][0-9]$/, {
 
 const zDayJs = z.custom<Dayjs>((val) => val instanceof dayjs, 'Invalid date');
 
+export const zUpsertEventFormData = z.object({
```
Comment (zaviermiller) [inline src/types/forms/events.ts:12]: we dont need to define a new type here. use the `EventFormData`

**Review by zaviermiller — DISMISSED** [review]
also it doesn't work.

**Comment by Frank-D-stein** [comment]
> also it doesn't work.

I am still waiting on the actual server function

File: src/views/EditEventView/index.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    const validationResult = validate(eventData);
+    if (validationResult) {
+      setValidationErrors(validationResult);
+      return;
+    }
+
+    // Clear the validation errors
+    setValidationErrors(undefined);
+
+    // NOTE: Actual submitting does not work yet, waiting on a dependancy
```
Comment (andlrutt) [inline src/views/EditEventView/index.tsx:35]:
```suggestion
    // NOTE: Actual submitting does not work yet, waiting on a dependency
```

**Review by andlrutt — DISMISSED** [review]

**Review by zaviermiller — DISMISSED** [review]
fix merge

**Review by zaviermiller — APPROVED** [review]

**Pull request merged**